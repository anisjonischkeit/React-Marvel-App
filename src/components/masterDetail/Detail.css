.detailView {
	/* width: 100%; */
	height: 100vh;
	overflow-y: auto;
	margin: 0;
	width: calc(100% - 400px);
	float: left;
}

.detailViewTopBar {
	display: none !important
}

@media only screen and (max-width: 960px) {
	.detailView {
		width: 100%;
		float: none;
	}

	.detailViewTopBar {
		display: flex !important
	}	
}