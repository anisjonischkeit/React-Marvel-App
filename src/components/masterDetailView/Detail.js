import React from 'react';
import './Detail.css'
import IconButton from 'material-ui/IconButton';
import BackIcon from 'material-ui/svg-icons/hardware/keyboard-arrow-left';

import AppBar from 'material-ui/AppBar'

export default props => (
  <div className='detailView'>
    <AppBar
      className='detailViewTopBar'
      title={props.title}
      iconElementLeft={<IconButton onClick={props.onBackClick}><BackIcon /></IconButton>}
      iconClassNameRight="muidocs-icon-action-home"
    />
    {props.children}
  </div>
)