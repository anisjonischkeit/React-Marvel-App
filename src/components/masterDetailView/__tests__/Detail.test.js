import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json'

import IconButton from 'material-ui/IconButton';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
import Detail from "../Detail"

describe("Detail Component", () => {
	let timesGoneBack = 0
	const backClickHandler = () => {
		timesGoneBack += 1
	}

	const detailComponent = mount(
		(<MuiThemeProvider>
			<Detail
				title="Sample Title"
				onBackClick={backClickHandler}
			>
				<p id="sampleChild">sampleChild</p>
				<p id="sampleChild2">sampleChild</p>
			</Detail>
		</MuiThemeProvider>)
	)

	describe("Renders correctly", () => {
		it("Matches Snapshot", () => {
			let tree = toJson(detailComponent);
			expect(tree).toMatchSnapshot();
		})

		it("Renders Children", () => {
			expect(detailComponent.find("#sampleChild").exists()).toBeTruthy()
			expect(detailComponent.find("#sampleChild2").exists()).toBeTruthy()
			expect(detailComponent.find("#sampleChild1000").exists()).toBeFalsy()
		})

	})
	
	describe("onClicks trigger Correctly", () => {
		it("onBackClick Triggers correctly", () => {
			const backBtnClick = detailComponent.find(IconButton).first().props().onClick
			expect(timesGoneBack).toBe(0)
			backBtnClick()
			expect(timesGoneBack).toBe(1)
			backBtnClick()
			expect(timesGoneBack).toBe(2)
			backBtnClick()
			backBtnClick()
			backBtnClick()
			expect(timesGoneBack).toBe(5)
		})
	})

})