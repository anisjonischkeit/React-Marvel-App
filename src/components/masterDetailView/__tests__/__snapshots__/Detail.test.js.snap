// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Detail Component Renders correctly Matches Snapshot 1`] = `
<MuiThemeProvider>
  <Component
    onBackClick={[Function]}
    title="Sample Title"
  >
    <div
      className="detailView"
    >
      <AppBar
        className="detailViewTopBar"
        iconClassNameRight="muidocs-icon-action-home"
        iconElementLeft={
          <IconButton
            disableTouchRipple={false}
            disabled={false}
            iconStyle={Object {}}
            onClick={[Function]}
            tooltipPosition="bottom-center"
            touch={false}
          >
            <HardwareKeyboardArrowLeft />
          </IconButton>
        }
        showMenuIconButton={true}
        title="Sample Title"
        zDepth={1}
      >
        <Paper
          circle={false}
          className="detailViewTopBar"
          rounded={false}
          style={
            Object {
              "backgroundColor": "#00bcd4",
              "display": "flex",
              "paddingLeft": 24,
              "paddingRight": 24,
              "position": "relative",
              "width": "100%",
              "zIndex": 1100,
            }
          }
          transitionEnabled={true}
          zDepth={1}
        >
          <div
            className="detailViewTopBar"
            style={
              Object {
                "WebkitTapHighlightColor": "rgba(0,0,0,0)",
                "backgroundColor": "#00bcd4",
                "borderRadius": "0px",
                "boxShadow": "0 1px 6px rgba(0, 0, 0, 0.12),
                       0 1px 4px rgba(0, 0, 0, 0.12)",
                "boxSizing": "border-box",
                "color": "rgba(0, 0, 0, 0.87)",
                "display": "flex",
                "fontFamily": "Roboto, sans-serif",
                "muiPrepared": true,
                "paddingLeft": 24,
                "paddingRight": 24,
                "position": "relative",
                "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                "width": "100%",
                "zIndex": 1100,
              }
            }
          >
            <div
              style={
                Object {
                  "marginLeft": -16,
                  "marginRight": 8,
                  "marginTop": 8,
                  "muiPrepared": true,
                }
              }
            >
              <IconButton
                disableTouchRipple={false}
                disabled={false}
                iconStyle={
                  Object {
                    "color": "#ffffff",
                    "fill": "#ffffff",
                  }
                }
                onClick={[Function]}
                tooltipPosition="bottom-center"
                touch={false}
              >
                <EnhancedButton
                  centerRipple={true}
                  containerElement="button"
                  disableTouchRipple={false}
                  disabled={false}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onKeyboardFocus={[Function]}
                  onMouseEnter={[Function]}
                  onMouseLeave={[Function]}
                  onMouseOut={[Function]}
                  onTouchStart={[Function]}
                  style={
                    Object {
                      "boxSizing": "border-box",
                      "fontSize": 0,
                      "height": 48,
                      "overflow": "visible",
                      "padding": 12,
                      "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                      "width": 48,
                    }
                  }
                  tabIndex={0}
                  type="button"
                >
                  <button
                    disabled={false}
                    onBlur={[Function]}
                    onClick={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    onKeyUp={[Function]}
                    onMouseEnter={[Function]}
                    onMouseLeave={[Function]}
                    onMouseOut={[Function]}
                    onTouchStart={[Function]}
                    style={
                      Object {
                        "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
                        "background": "none",
                        "border": 10,
                        "boxSizing": "border-box",
                        "cursor": "pointer",
                        "display": "inline-block",
                        "fontFamily": "Roboto, sans-serif",
                        "fontSize": 0,
                        "fontWeight": "inherit",
                        "height": 48,
                        "margin": 0,
                        "muiPrepared": true,
                        "outline": "none",
                        "overflow": "visible",
                        "padding": 12,
                        "position": "relative",
                        "textDecoration": "none",
                        "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                        "verticalAlign": null,
                        "width": 48,
                      }
                    }
                    tabIndex={0}
                    type="button"
                  >
                    <TouchRipple
                      abortOnScroll={true}
                      centerRipple={true}
                      key="touchRipple"
                    >
                      <div
                        onMouseDown={[Function]}
                        onMouseLeave={[Function]}
                        onMouseUp={[Function]}
                        onTouchEnd={[Function]}
                        onTouchStart={[Function]}
                      >
                        <HardwareKeyboardArrowLeft
                          key=".0"
                          style={
                            Object {
                              "color": "#ffffff",
                              "fill": "#ffffff",
                            }
                          }
                        >
                          <HardwareKeyboardArrowLeft
                            style={
                              Object {
                                "color": "#ffffff",
                                "fill": "#ffffff",
                              }
                            }
                          >
                            <SvgIcon
                              onMouseEnter={[Function]}
                              onMouseLeave={[Function]}
                              style={
                                Object {
                                  "color": "#ffffff",
                                  "fill": "#ffffff",
                                }
                              }
                              viewBox="0 0 24 24"
                            >
                              <svg
                                onMouseEnter={[Function]}
                                onMouseLeave={[Function]}
                                style={
                                  Object {
                                    "color": "#ffffff",
                                    "display": "inline-block",
                                    "fill": "#ffffff",
                                    "height": 24,
                                    "muiPrepared": true,
                                    "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                                    "userSelect": "none",
                                    "width": 24,
                                  }
                                }
                                viewBox="0 0 24 24"
                              >
                                <path
                                  d="M15.41 16.09l-4.58-4.59 4.58-4.59L14 5.5l-6 6 6 6z"
                                />
                              </svg>
                            </SvgIcon>
                          </HardwareKeyboardArrowLeft>
                        </HardwareKeyboardArrowLeft>
                      </div>
                    </TouchRipple>
                  </button>
                </EnhancedButton>
              </IconButton>
            </div>
            <h1
              onClick={[Function]}
              style={
                Object {
                  "boxFlex": 1,
                  "color": "#ffffff",
                  "flex": "1",
                  "fontSize": 24,
                  "fontWeight": 400,
                  "height": 64,
                  "letterSpacing": 0,
                  "lineHeight": "64px",
                  "margin": 0,
                  "muiPrepared": true,
                  "overflow": "hidden",
                  "paddingTop": 0,
                  "textOverflow": "ellipsis",
                  "whiteSpace": "nowrap",
                }
              }
            >
              Sample Title
            </h1>
            <IconButton
              disableTouchRipple={false}
              disabled={false}
              iconClassName="muidocs-icon-action-home"
              iconStyle={
                Object {
                  "color": "#ffffff",
                  "fill": "#ffffff",
                }
              }
              onClick={[Function]}
              style={
                Object {
                  "marginLeft": "auto",
                  "marginRight": -16,
                  "marginTop": 8,
                }
              }
              tooltipPosition="bottom-center"
              touch={false}
            >
              <EnhancedButton
                centerRipple={true}
                containerElement="button"
                disableTouchRipple={false}
                disabled={false}
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onKeyboardFocus={[Function]}
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
                onMouseOut={[Function]}
                onTouchStart={[Function]}
                style={
                  Object {
                    "boxSizing": "border-box",
                    "fontSize": 0,
                    "height": 48,
                    "marginLeft": "auto",
                    "marginRight": -16,
                    "marginTop": 8,
                    "overflow": "visible",
                    "padding": 12,
                    "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                    "width": 48,
                  }
                }
                tabIndex={0}
                type="button"
              >
                <button
                  disabled={false}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseEnter={[Function]}
                  onMouseLeave={[Function]}
                  onMouseOut={[Function]}
                  onTouchStart={[Function]}
                  style={
                    Object {
                      "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
                      "background": "none",
                      "border": 10,
                      "boxSizing": "border-box",
                      "cursor": "pointer",
                      "display": "inline-block",
                      "fontFamily": "Roboto, sans-serif",
                      "fontSize": 0,
                      "fontWeight": "inherit",
                      "height": 48,
                      "margin": 0,
                      "marginLeft": "auto",
                      "marginRight": -16,
                      "marginTop": 8,
                      "muiPrepared": true,
                      "outline": "none",
                      "overflow": "visible",
                      "padding": 12,
                      "position": "relative",
                      "textDecoration": "none",
                      "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                      "verticalAlign": null,
                      "width": 48,
                    }
                  }
                  tabIndex={0}
                  type="button"
                >
                  <TouchRipple
                    abortOnScroll={true}
                    centerRipple={true}
                    key="touchRipple"
                  >
                    <div
                      onMouseDown={[Function]}
                      onMouseLeave={[Function]}
                      onMouseUp={[Function]}
                      onTouchEnd={[Function]}
                      onTouchStart={[Function]}
                    >
                      <FontIcon
                        className="muidocs-icon-action-home"
                        color="rgba(0, 0, 0, 0.87)"
                        onMouseEnter={[Function]}
                        onMouseLeave={[Function]}
                        style={
                          Object {
                            "color": "#ffffff",
                            "fill": "#ffffff",
                          }
                        }
                      >
                        <span
                          className="muidocs-icon-action-home"
                          color="rgba(0, 0, 0, 0.87)"
                          onMouseEnter={[Function]}
                          onMouseLeave={[Function]}
                          style={
                            Object {
                              "color": "#ffffff",
                              "display": "inline-block",
                              "fill": "#ffffff",
                              "fontSize": 24,
                              "muiPrepared": true,
                              "position": "relative",
                              "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                              "userSelect": "none",
                            }
                          }
                        />
                      </FontIcon>
                    </div>
                  </TouchRipple>
                </button>
              </EnhancedButton>
            </IconButton>
          </div>
        </Paper>
      </AppBar>
      <p
        id="sampleChild"
      >
        sampleChild
      </p>
      <p
        id="sampleChild2"
      >
        sampleChild
      </p>
    </div>
  </Component>
</MuiThemeProvider>
`;
